compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -2736,6 +2739,10 @@ object Types {
       case that: ThisType => tref.eq(that.tref)
       case _ => false
     }
+
+    def sameThis(that: Type)(using Context): Boolean = (that eq this) || that.match
+      case that: ThisType => this =:= that
```

Comment:
=:= between two ThisType just compares their `cls` field, so we could inline that here.